bank-sdk/example-app/src/androidTest/java/net/gini/android/bank/sdk/exampleapp/ui/screens/DigitalInvoiceScreen.kt

```diff
@@ -87,16 +91,23 @@ class DigitalInvoiceScreen {
         return this
     }
 
-    fun checkForReturnReasonsList() {
-        onView(withText(net.gini.android.bank.sdk.R.string.gbs_digital_invoice_return_reason_dialog_title)).check(
-            matches(isDisplayed()))
+    fun checkForReturnReasonsList(): Boolean {
+        var isReturnReasonDisplayed = false
+        onView(withText(net.gini.android.bank.sdk.R.string.gbs_digital_invoice_return_reason_dialog_title))
+            .check { view, noViewFoundException ->
+                if (noViewFoundException == null || view.isShown()) {
+                    isReturnReasonDisplayed = true
+                }
+            }
+        return isReturnReasonDisplayed
     }
 
-    fun  checkItemCountOnReturnReasonsList() {
+    fun  checkItemCountOnReturnReasonsList(): Int{
         val uiCollection =
             UiCollection(UiSelector().className("android.widget.ListView"))
         val itemSize = uiCollection.childCount
         onView(withClassName(`is`("android.widget.ListView"))).check(matches(hasChildCount(itemSize)))
```

Comment:
[Clarification] Do we need to perform this check here? Is it not enough to return `itemSize` and check it in the main test case?

if so, could we rename the method to something like `returnItemCountOnReturnReasonsList()` ?

Reply:
I think we have to perform the check here because the result of the check is set as `itemSize` which is then be return to main test..
But I have renamed the function as suggested.